core/primitives/src/rpc.rs

```diff
@@ -8,4 +9,13 @@ pub struct RpcQueryRequest {
     pub block_id: MaybeBlockId,
     #[serde(flatten)]
     pub request: QueryRequest,
+    pub finality: Finality,
+}
+
+/// Different types of finality.
+#[derive(Serialize, Deserialize, BorshSerialize, BorshDeserialize, Clone, Debug, PartialEq, Eq)]
+pub enum Finality {
```

Comment:
Given we use this structure not only in RPC, it is not the right place to put it. Let's move it into `views.rs`.